cmd/kubeadm/app/cmd/phases/upgrade/addons.go

```diff
@@ -62,6 +67,21 @@ func NewAddonPhase() workflow.Phase {
 }
 
 func shouldUpgradeAddons(client clientset.Interface, cfg *kubeadmapi.InitConfiguration, out io.Writer) (bool, error) {
+	nodeSelector := labels.SelectorFromSet(labels.Set(map[string]string{
```

Comment:
in other places we check if /etc/kubernetes/manifests/kube-apiserver.yaml exists to say this is a cp node

Reply:
We should implement `IsControlPlaneNode() bool` for both `kubeadm upgrade apply` and `kubeadm upgrade node`

https://github.com/kubernetes/kubernetes/blob/master/cmd/kubeadm/app/cmd/phases/upgrade/node/data.go#L29

https://github.com/kubernetes/kubernetes/blob/f836773540cc83488250a12637e7d924078ef284/cmd/kubeadm/app/cmd/upgrade/node.go#L150-L154

Reply:
Then we can use `data.IsControlPlaneNode()`

just like :
```go
		// If this is not a control-plane node, this phase should not be executed.
		if !data.IsControlPlaneNode() {
			fmt.Println("[upgrade/kubeconfig] Skipping phase. Not a control plane node.")
			return nil
		}
```

Reply:
I can help send a fix if needed.

Reply:
> Then we can use `data.IsControlPlaneNode()`

IIRC, the node data doesn't implement this.


> I can help send a fix if needed.

OK.  Then I would not update this PR.

Reply:
https://github.com/kubernetes/kubernetes/pull/127289